src/Interpreters/IInterpreterUnionOrSelectQuery.h

```diff
@@ -40,6 +40,8 @@ class IInterpreterUnionOrSelectQuery : public IInterpreter
 
     void extendQueryLogElemImpl(QueryLogElement & elem, const ASTPtr &, ContextPtr) const override;
 
+    bool usesViewSource() { return uses_view_source; }
```

Comment:
Well, it's really hard do understand what this flag is needed for. At least, we need a comment.

From what I can see, this flag should be `true` if storage, or some of joined storages, or some subquery uses `view_source` (which is not a real storage, but a virtual one, which returns only a new inserted block). Ans this is needed to know if we inserting into MV now, so that we don't remember scalars which is calculated on top of incomplete table.

Can we specify this flag externally? Like, add a separate option to `SelectQueryOptions` which tells that we should use only local scalars. And set it here:
https://github.com/ClickHouse/ClickHouse/blob/67fc9d7129f607f9551bb5ab9944d8bc23712ddc/src/Processors/Transforms/buildPushingToViewsChain.cpp#L440

Reply:
Ok, probably, we can't just do it this way (or can, but it will be not so well) because some scalars from inside MV may not depend on `view_source`, and so we can use them as global.

Reply:
Still, comment is needed.

Reply:
> Well, it's really hard do understand what this flag is needed for. At least, we need a comment.

Absolutely. Will do.

> Ok, probably, we can't just do it this way (or can, but it will be not so well) because some scalars from inside MV may not depend on view_source, and so we can use them as global.

Yes, that's the reason behind the flag. Be able to know if the query can be cached globally or only locally to the MV block.